server/src/main/java/com/google/coffeehouse/common/Person.java

```diff
@@ -0,0 +1,120 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.coffeehouse.common;
+
+import com.google.coffeehouse.util.IdentifierGenerator;
+import com.google.coffeehouse.util.UuidWrapper;
+import java.util.Optional; 
+
+/**
+ * Person is the class than encapsulates the information associated with a user who has an account.
+ * 
+ * <p>It implementes the {@link Saveable} interface because it will be able to save itself
+ * in the database.
+ */
+public class Person implements Saveable {
+  private String nickname;
+  private String email;
+  private String pronouns;
+  private String userId;
+
+  private Person(Builder builder) {
+    this.nickname = builder.nickname;
+    this.email = builder.email;
+    this.pronouns = builder.pronouns;
+    this.userId = builder.idGenerator.generateId();
+  }
+
+  public String getNickname() {
+    return nickname;
+  }
+
+  public String getEmail() {
+    return email;
+  }
+
+  /** Returns an optional because the pronouns may, or may not, exist. */
+  public Optional<String> getPronouns() {
+    return pronouns == null ? Optional.empty() : Optional.of(pronouns);
+  }
+
+  public String getUserId() {
+    return userId;
+  }
+
+  public void setPronouns(String pronouns) {
+    this.pronouns = pronouns;
+  }
+
+  public void setEmail(String email) {
+    this.email = email;
+  }
+
+  public void setNickname(String nickname) {
+    this.nickname = nickname;
+  }
```

Comment:
In what cases do you expect these setters to be used instead of the builder?

Reply:
I assumed that the setters would be used when modifying an Person (a user changes their pronouns), whereas the builder would only create the person. This is the reason why I didn't include a setUserId method, it should only be set once, when the object is made.